### PR TITLE
Log#add: ensure reasons to be array of strings

### PR DESCRIPTION
**Pull request by rodweb**
Currently, adding an action with `reasons` set to something other than array prevents `Log#removeReason` from working properly, as `MemoryStore#removeReason` expects `meta.reasons` to be an array. This PR ensures whatever data passed as `reasons` to be converted to an array.

### CODE REVIEW

File: test/log.test.js [hunk 1]
```diff
@@ -476,3 +476,20 @@ it('removes reasons when keepLast option is used', function () {
...
+    expect(meta.reasons).toEqual([])
+    return log.add({ type: '2' }, { reasons: null })
+  }).then(function (meta) {
+    expect(meta.reasons).toEqual([])
+    return log.add({ type: '3' }, { reasons: 'a' })
+  }).then(function (meta) {
+    expect(meta.reasons).toEqual(['a'])
+    return log.add({ type: '4' }, { reasons: [null, false, 1, { foo: 'bar' }] })
+  }).then(function (meta) {
+    expect(meta.reasons).toEqual(['null', 'false', '1', '[object Object]'])
```
Comment (ai) [inline test/log.test.js]: I think `null` or `[object Object]` reasons are unexpected. Let's throw a error on non-string reason.

**Comment by ai** [comment]
Hm, why do you use `reasons.map` instead of `reasons.forEach` to check reasons type. Map creates new array, so JS will can Garbage Collector more often.

**Comment by ai** [comment]
Also could you try to make error text a little shorter? Logux Core is used in Logux Client which is critical for size.

**Comment by rodweb** [comment]
> Hm, why do you use reasons.map instead of reasons.forEach to check reasons type. Map creates new array, so JS will can Garbage Collector more often.

Good point, will replace `map` to `forEach`.

> Also could you try to make error text a little shorter?

`Expected "reasons" to contain strings`?
`"reasons" must contain strings`?
What do you think?

**Comment by ai** [comment]
Or `Expected "reasons" to be strings`

**Pull request merged**